app/uk/gov/hmrc/vulnerabilities/service/UpdateVulnerabilitiesService.scala

```diff
@@ -38,26 +39,55 @@ class UpdateVulnerabilitiesService @Inject()(
 )(implicit ec: ExecutionContext) extends
   Logging {
 
-  def updateVulnerabilities()(implicit hc: HeaderCarrier): Future[Unit] = {
+  def updateAllVulnerabilities()(implicit hc: HeaderCarrier): Future[Unit] = {
     for {
-      wrw             <- releasesConnector.getCurrentReleases()
-      svDeps           = whatsRunningWhereService.getEnvsForServiceVersion(wrw)
-      //Only download reports that don't exist in last 7 days in our raw reports collection
+      svDeps          <- getCurrentServiceDependencies
+      //Only download reports that don't exist in last 24 hours in our raw reports collection
       recentReports   <- rawReportsRepository.getReportsInLastXDays()
       outOfDateSVDeps  = whatsRunningWhereService.removeSVDIfRecentReportExists(svDeps, recentReports)
-      _               <- xrayService.processReports(outOfDateSVDeps)
-      _                = logger.info(s"Finished generating and inserting reports into the rawReports collection")
+      _               <- processXrayReports(outOfDateSVDeps)
+      _               <- updateVulnerabilitySummaries(svDeps)
+    } yield ()
+  }
+
+
+  def updateVulnerabilities(serviceName: String,
+                            version: String,
+                            environment: String)(implicit hc: HeaderCarrier): Future[Unit] = {
+    val svDepsToUpdate = Seq(ServiceVersionDeployments(serviceName, version, Seq(environment)))
```

Comment:
could be moved into the for-comprehension, e.g
`for { svDepsToUpdate <- Future.successful(Seq(ServiceVersionDeployments(serviceName, version, Seq(environment))))` (then can nuke the additional `{` wrapping the function)